python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -365,6 +368,19 @@ def _check_getter(self):
             raise AttributeError('cannot modify or delete "%s" property (declared as "getter")' % _getName(self))
 
     def __get__(self, obj, cls):
+
+        self_name = _getName(self)
+        GangaObject._acquireChildLock(self_name, obj)
```

Comment:
```python
import threading
l = threading.RLock()
with l:
    <do stuff>
```
[locking contexts](https://docs.python.org/2/library/threading.html#using-locks-conditions-and-semaphores-in-the-with-statement)